### PR TITLE
[lapack-reference|fortran] trying out fortran (just testing)

### PR DESCRIPTION
**Pull request by Neumann-A**
continuation of #11121

~~requires msys fixes first???~~

### CODE REVIEW

File: scripts/toolchains/windows_fortran_rules.cmake [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#endif()
+
+# link a fortran program
+if(NOT CMAKE_Fortran_LINK_EXECUTABLE)
+  #set(CMAKE_Fortran_LINK_EXECUTABLE
+  #  "<CMAKE_Fortran_COMPILER> <CMAKE_Fortran_LINK_FLAGS> <LINK_FLAGS> <FLAGS> <OBJECTS>  -o <TARGET> <LINK_LIBRARIES>")
+endif()
+
+if(CMAKE_Fortran_STANDARD_LIBRARIES_INIT)
+  set(CMAKE_Fortran_STANDARD_LIBRARIES "${CMAKE_Fortran_STANDARD_LIBRARIES_INIT}"
```
Comment (Voskrese) [inline scripts/toolchains/windows_fortran_rules.cmake:121]: set(CMAKE_Fortran_STANDARD_LIBRARIES "${CMAKE_Fortran_STANDARD_LIBRARIES_INIT}" "-lgfortran" # gnu fortran ?

Comment (Neumann-A) [reply scripts/toolchains/windows_fortran_rules.cmake:121]: http://www.netlib.org/lapack/lawnspdf/lawn270.pdf

Need to probably make it dynamic

**Comment by Neumann-A** [comment]
@BillyONeal: have the msys issues been solved?

**Comment by Neumann-A** [comment]
@BillyONeal @JackBoosY: Somebody forgot to activate VS on UWP CI?

```
  It fails with the following output:

    Change Dir: C:/agent/_work/1/s/buildtrees/lapack-reference/x64-uwp-rel/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/devenv.com CMAKE_TRY_COMPILE.sln /build Debug /project cmTC_7122a && 
    Microsoft Visual Studio 2019 Version 16.5.4.
    Copyright (C) Microsoft Corp. All rights reserved.
    
    The license for Visual Studio has expired.
    
    The evaluation period for this product has ended.
    
```

**Comment by Neumann-A** [comment]
In general seems to work findLapack seems to have some minor problems in one buildtype

**Comment by Neumann-A** [comment]
Hmm tested on of the x64-windows-static build failures and was unable to reproduce it. 
```

Building package openmvg[core]:x64-windows-static... done
Installing package openmvg[core]:x64-windows-static...
Installing package openmvg[core]:x64-windows-static... done
Elapsed time for package openmvg:x64-windows-static: 3.684 min

Total elapsed time: 7.469 min

The package openmvg:x64-windows-static provides CMake targets:

    find_package(OpenMVG CONFIG REQUIRED)
    # Note: 17 target(s) were omitted.
    target_link_libraries(main PRIVATE OpenMVG::vlsift OpenMVG::openMVG_sfm OpenMVG::openMVG_exif OpenMVG::openMVG_fast)
```

Both Release and Debug builds find Lapack.

**Pull request closed without merging**